src/main/java/frc/robot/commands/AutonModes.java

```diff
@@ -0,0 +1,198 @@
+    this.shooterSubsystem = s;
+    this.limelightSubsystem = l;
+    this.cdsSubsystem = c;
+    this.intakeSubsystem = i;
+
+    // parameters are route names (names based on pathweaver, they are located under Auton file)
+    taxiTrajectories = getTrajectories("Taxi");
+    oneBallTrajectories = getTrajectories("OneBall");
+    twoBallTrajectories = getTrajectories("TwoBall");
+    threeBallTrajectories = getTrajectories("ThreeBall");
+    fourBallTrajectories = getTrajectories("FourBall");
+
+    // create ramsete commands using the trajectories
+    taxiRamseteCommands = getRamseteCommands(taxiTrajectories);
+    oneBallRamseteCommands = getRamseteCommands(oneBallTrajectories);
+    twoBallRamseteCommands = getRamseteCommands(twoBallTrajectories);
+    threeRamseteCommands = getRamseteCommands(threeBallTrajectories);
+    fourRamseteCommands = getRamseteCommands(fourBallTrajectories);
+  }
+
+  private void initializeCommandGroups() {
+    // TODO: the wait time should not be a constant, should be configurable
+    taxiCommand =
+        new SequentialCommandGroup(
+            new WaitCommand(Constants.delaytaxi),
+            taxiRamseteCommands[0].beforeStarting(
+                () -> driveBaseSubsystem.resetOdometry(taxiTrajectories[0].getInitialPose())));
+
+    oneBallCommand =
+        new SequentialCommandGroup(
+            new WaitCommand(Constants.delayshot),
+            // new ShooterPrime(shooterSubsystem, limelightSubsystem, cdsSubsystem),
+            new ShooterPressed(
+                shooterSubsystem,
+                limelightSubsystem,
+                cdsSubsystem,
+                true), // also has a time delay of 2-3 seconds
+            new WaitCommand(Constants.delaytaxi),
+            oneBallRamseteCommands[0].beforeStarting(
+                () -> driveBaseSubsystem.resetOdometry(taxiTrajectories[0].getInitialPose())));
+
+    twoBallParallel =
+        new ParallelDeadlineGroup(
+            twoBallRamseteCommands[0].beforeStarting(
+                () -> driveBaseSubsystem.resetOdometry(twoBallTrajectories[0].getInitialPose())),
+            new IntakeForwardCommand(intakeSubsystem));
+    twoBallCommand =
+        new SequentialCommandGroup(
+            new WaitCommand(Constants.delaytaxi),
+            twoBallParallel,
+            twoBallRamseteCommands[1],
+            new WaitCommand(Constants.delayshot),
+            // new ShooterPrime(shooterSubsystem, limelightSubsystem, cdsSubsystem)
+            new ShooterPressed(shooterSubsystem, limelightSubsystem, cdsSubsystem, true));
+  }
+
+  private Trajectory[] getTrajectories(String routeString) {
+    // method to intialize multiple trajectories in one auton route
+    String routeName = routeString; // TODO: replace route for the correct one on pathweaver
+    File f =
+        new File(
+            "Autos/" + routeName); // get file with all the names of the path for the auton routine
+    Scanner fileScanner = null;
```

Comment:
No need to assign to null
```suggestion
    Scanner fileScanner;
```